### PR TITLE
KyselovaMaria

### PR DESCRIPTION
**Pull request by KyselovaMaria**
Pull request for Titanic

### CODE REVIEW

**Review by Dmitriy-NK — COMMENTED** [review]
File structure is OK. No additional change is required.

**Review by Dmitriy-NK — COMMENTED** [review]
1. График 4 (в большом наборе графиков) построен с нормированием к полному числу пассажиров. Предполагается сначала разбиение на подмножества а потом для каждого подмножества построение графика. у seaborn.kdeplot для этого используется аргумент "common_norm".
2. Некорректно отрисован график, для гендерного распределения. Подставленные значения не соответствуют делению.
3. В разделе "graph down by gender" отсутствует график пропорционального распределения "# Next adjust graph to display the proportions of survival by gender" и просьба добавить несколько предложений по своей оценке после "Add your summary about recieved data for raw value counts and proportionally"
4. Выводы по Pclass больше подходят предыдущему разделу.
5. Kaggle уже предлагает разбиение данных  на тестовую и тренировочную выборку. Просьба переделать и сделать оценку качества работы моделей на тестовом датасете. 
6. Приведите confusion matrix к заданному в описании формату(включая нормализацию)
>При построении ROC кривой предпочтительно делать график квадратным

**Comment by KyselovaMaria** [comment]
Я добавила новый файл с изменениями по указаным выше замечаниям.

**Review by Dmitriy-NK — COMMENTED** [review]
**Не нужно добавлять новый файл с изменениями, правильно будет заменить текущий**

для 3-го пункта графика нет, нужен график нормированного(в долях единицы) распределения выживших в двух подмножествах 'female','male'.
5-й пункт также проигнорирован, натренируйте данные на всех доступных данных из файла train.csv а оценку проводите на данных из test.csv. train_test_split в данном задании использовать не нужно.
касательно confusion matrix - по умолчанию sklearn.metrics.confusion_matrix считает метку 0 - положительной (это связано с возможностью классификации с многими метками) так собственный порядок можно задать в аргументе labels, в нашем случае [1,0]  [detailes](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.confusion_matrix.html)

**Review by Dmitriy-NK — COMMENTED** [review]
Также просьба для нескольких графиков подряд использовать вариант предложенный в начале задания с их комбинированием  в одну панель, это значительно улучшает читаемость и дает возможность анализа сразу нескольких зависимостей.

**Review by Dmitriy-NK — COMMENTED** [review]
3-й пункт не изменился. Добавьте  диаграмму с двумя подмножествами male', 'female' на которой будет указана часть выживших и не выживших для каждого подмножества в долях единицы. [пример](https://github.com/srk-practicum/2022-knu-yolo/issues/35)
Что касается Confusion matrix, вопросы не к её внешнему оформлению, а к тому что перепутаны метки Survive и Non-Survive. И вместо:

  |   | Predict |   
-- | -- | -- | -- 
  |   | Survive | Non-Survive
Actual | Survive | .. | ..
  | Non-Survive | .. | ..

Ваша матрица хранит значения так:

  |   | Predict |   
-- | -- | -- | -- 
  |   | Non-Survive | Survive
Actual | Non-Survive | .. | ..
  | Survive | .. | ..

**Review by Dmitriy-NK — COMMENTED** [review]
Только сейчас обратил внимание: 3-й график "Class Distribution" в наборе графиков   построен с ошибкой

**Review by Dmitriy-NK — APPROVED** [review]
LGTM
